crypto/ec/ec_key.c

```diff
@@ -482,8 +482,10 @@ int EC_KEY_set_private_key(EC_KEY *key, const BIGNUM *priv_key)
      * might temporarily overflow the order length.
      */
     tmp_key = BN_dup(priv_key);
-    if (tmp_key == NULL)
+    if (tmp_key == NULL) {
+        key->priv_key = NULL;
```

Comment:
The old priv_key needs to be freed

Reply:
I would actually recommend to keep L485-L488 as they were, and add a new `if` block around L445, something similar to this:

~~~c
if (priv_key == NULL) {
    /**
     * If passed NULL, we properly clear `key->priv_key`.
     * We return `0` to comply with legacy behavior for this function,
     * see <https://github.com/openssl/openssl/issues/18744#issuecomment-1195175696>.
     */
    BN_clear_free(key->priv_key);
    key->priv_key = NULL;
    return 0; /* intentional for legacy compatibility */
}
~~~